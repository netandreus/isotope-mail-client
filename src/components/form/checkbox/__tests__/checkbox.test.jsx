import React from 'react';
import {shallow} from 'enzyme/build/index';
import Checkbox from '../checkbox';

describe('Checkbox component test suite', () => {
  test('Snapshot render, should render checkbox', () => {
    const {id, fieldClass, inputClass, required, onChange, checked} = {
      id: 'Bourne', fieldClass: 'field-class', inputClass: 'input-class',
      required: true, onChange: () => {}, checked: true};
    const checkbox = shallow(<Checkbox id={id} fieldClass={fieldClass} inputClass={inputClass} required={required}
      onChange={onChange} checked={checked} />);
    expect(checkbox).toMatchSnapshot();
  });
  test('click/change, should trigger onChange function', () => {
    // Given
    const onChange = jest.fn();
    const checkbox = shallow(<Checkbox onChange={onChange}/>);

    // When
    checkbox.find('.mdc-checkbox input').simulate('change', {target: {checked: true}});

    // Then
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
