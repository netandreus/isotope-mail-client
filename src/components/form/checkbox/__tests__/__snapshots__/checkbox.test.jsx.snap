// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component test suite Snapshot render, should render checkbox 1`] = `
<div
  className="mdc-checkbox mdc-checkbox--primary
        field-class"
>
  <input
    checked={true}
    className="mdc-checkbox__native-control input-class"
    id="Bourne"
    onChange={[Function]}
    required={true}
    type="checkbox"
  />
  <div
    className="mdc-checkbox__background"
  >
    <svg
      className="mdc-checkbox__checkmark"
      viewBox="0 0 24 24"
    >
      <path
        className="mdc-checkbox__checkmark-path"
        d="M1.73,12.91 8.1,19.28 22.79,4.59"
        fill="none"
      />
    </svg>
    <div
      className="mdc-checkbox__mixedmark"
    />
  </div>
</div>
`;
